lib/rex/proto/teamcity/rsa.rb

```diff
@@ -0,0 +1,82 @@
+module Rex::Proto::Teamcity::Rsa
+  # https://github.com/openssl/openssl/blob/a08a145d4a7e663dd1e973f06a56e983a5e916f7/crypto/rsa/rsa_pk1.c#L125
+  # https://datatracker.ietf.org/doc/html/rfc3447#section-7.2.1
+  def self.pkcs1pad2(text, n)
+    if n < text.length + 11
+      raise ArgumentError, 'Message too long'
+    end
+
+    r = Array.new(n, 0)
+    n -= 1
+    r[n] = text.length
+
+    i = text.length - 1
+
+    while i >= 0 && n > 0
+      c = text[i].ord
+      i -= 1
+      n -= 1
+      r[n] = c % 0x100
+    end
+    n -= 1
+    r[n] = 0
+
+    while n > 2
+      n -= 1
+      r[n] = SecureRandom.random_bytes(1).bytes.first
+    end
+
+    n -= 1
+    r[n] = 2
+    n -= 1
+    r[n] = 0
+
+    r.pack("C*").unpack1("H*").to_i(16)
+  end
+
+  # @param [String] modulus
+  # @param [String] exponent
+  # @param [String] text
+  # @return [String]
+  def self.rsa_encrypt(modulus, exponent, text)
+    n = modulus.to_i(16)
+    e = exponent.to_i(16)
+
+    padded_as_big_int = pkcs1pad2(text, (n.bit_length + 7) >> 3)
+    encrypted = padded_as_big_int.to_bn.mod_exp(e, n)
+    h = encrypted.to_s(16)
+
+    h.length.odd? ? h.prepend('0') : h
+  end
+
+  def self.two_byte_chars?(str)
+    str.bytesize > str.length
```

Comment:
I don't know if this will have the same semantics as JavaScript if our input is being forced as ascii

https://github.com/rapid7/metasploit-framework/blob/9e0b0f565f666eadcd1c73911a4a63aa726d58ea/lib/metasploit/framework/common_engine.rb#L30-L32